File: flibrary/cdot.f
      function cdot(x,y,n)

c     Compute the inner product
c     dot=(x,y) for complex x,y
         implicit none
         integer n,i
         complex*16  x(n), y(n), cdot, val
        
         val=(0.0d0,0.d0)
         do i=1,n 
            val = val +dconjg( x(i))*y(i)
         enddo

         cdot=val

         return
         end

File: supocs/v1/cdot.cpp
#include "su.h"
#include "segy.h"
#include "Complex.h"

complex cdot(complex *x,complex *y,int n)
{
  //     Compute the inner product
  //     dot=(x,y) for complex x,y
        
   int i;
   complex  cdot, val;
   val.r=0;
   val.i=0;
   for (i=0;i<n;i++)  
       val = val +conjg(x[i])*y[i];
   cdot=val;
   return(val);
}





